### PR TITLE
[TX] Add remaining platform tx to caminoExecutor (spend / lock)

### PR DESCRIPTION
**Pull request by c4t-ag**
# Camino TX / spend adjustment
PR implements all platformTx from utxoset which use getMinimumSpendable.
Instead the JS based UTXO logic we let camino-node / go evaluate the resulting UTXOs.

### CODE REVIEW

**Review by Chriserus — APPROVED** [review]
LGTM

**Comment by c4t-ag** [comment]
I was not happy because of all the code changes in UTXO and API, so I reworked platform API:
-> move createTx functions from UTXO into new Builder (which can now either get old or new getMinimumSpendable from interface.
-> Create a Builder class which capsulates the spend call into camino-node.

Pls. @Chriserus review again. I'll merge earliest when evgeni has approves the chnages made in caminogo to you can verify that tx work as expected (e.g. RegisterNodeTx, AddValidatorTx, ImportTx and ExportTx).

File: src/apis/platformvm/api.ts [hunk 1]
```diff
@@ -2077,15 +2129,28 @@ export class PlatformVMAPI extends JRPCAPI {
...
+    toThreshold: number,
+    change: string[],
+    changeThreshold: number,
+    lockMode: LockMode,
     amountToLock: BN,
     amountToBurn: BN,
     encoding?: string
   ): Promise<SpendReply> => {
+    if (!["Unlocked", "Deposit", "Bond"].includes(lockMode)) {
+      throw new ProtocolError("Error -- PlatformAPI.spend: invalid lockeMode")
```
Comment (Chriserus) [inline src/apis/platformvm/api.ts]: Typo, `lockMode`, not `lockeMode`

File: src/apis/evm/api.ts [hunk 2]
```diff
@@ -638,12 +638,13 @@ export class EVMAPI extends JRPCAPI {
    * @param amount The amount being exported as a {@link https://github.com/indutny/bn.js/|BN}
    * @param assetID The asset id which is being sent
    * @param destinationChain The chainid for where the assets will be sent.
-   * @param toAddresses The addresses to send the funds
-   * @param fromAddresses The addresses being used to send the funds from the UTXOs provided
-   * @param changeAddresses The addresses that can spend the change remaining from the spent UTXOs
-   * @param asOf Optional. The timestamp to verify the transaction against as a {@link https://github.com/indutny/bn.js/|BN}
+   * @param fromAddressesHex The addresses to send the funds (hex)
+   * @param fromAddressesBech The addresses being used to send the funds from the UTXOs provided
+   * @param toAddresses An array of addresses as {@link https://github.com/feross/buffer|Buffer} who recieves the AVAX
```
Comment (Chriserus) [inline src/apis/evm/api.ts:643]: `AVAX`, I think it's `CAM`

File: tests/apis/platformvm/api.test.ts [hunk 3]
```diff
@@ -1403,42 +1407,43 @@ describe("PlatformVMAPI", (): void => {
...
+        startTime,
+        endTime,
+        PlatformVMConstants.MINSTAKE,
+        platformvm.getFee(),
+        assetID,
+        new UTF8Payload("hello world").getPayload(), UnixNow()
+      );
+      expect(txu2.toBuffer().toString('hex')).toBe(txu1.toBuffer().toString('hex'));
+      expect(txu2.toString()).toBe(txu1.toString());
+    });
```
Comment (Chriserus) [inline tests/apis/platformvm/api.test.ts:1444]: Just confirming, do we want this test commented out?

**Review by Chriserus — APPROVED** [review]

File: tests/apis/platformvm/api.test.ts (hunk 3 above)
Comment (c4t-ag) [reply tests/apis/platformvm/api.test.ts:1444]: It was commented out before, just formatting

File: src/apis/evm/api.ts (hunk 2 above)
Comment (c4t-ag) [reply src/apis/evm/api.ts:643]: Ideal solution would be "Native Asset" all over the places but we want to be less invasive.
Beside this the comment was wrong and simply copied from the implementation.
Pls. always keep in mind that caminojs is capable for different native assets, also for AVAX.

**Pull request merged**